logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreFailure.kt

```diff
@@ -82,10 +84,14 @@ internal inline fun <T : Any> wrapApiRequest(networkCall: () -> NetworkResponse<
         is NetworkResponse.Error -> {
             kaliumLogger.e(result.kException.stackTraceToString())
             val exception = result.kException
-            if (exception is KaliumException.GenericError && exception.cause is IOException) {
-                Either.Left(NetworkFailure.NoNetworkConnection(exception))
+            if (exception.cause?.message?.contains("socks", true) == true) {
```

Comment:
Could you please extract the hard-coded text and also set the variables names? and also you can remove the `== true` section from the `if` like: `if (exception.cause?.message?.contains("socks", true))`